src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/ExtractToNewComponentCodeActionProvider.cs

```diff
@@ -47,51 +51,169 @@ public Task<ImmutableArray<RazorVSInternalCodeAction>> ProvideAsync(RazorCodeAct
             return SpecializedTasks.EmptyImmutableArray<RazorVSInternalCodeAction>();
         }
 
+        var (startElementNode, endElementNode) = GetStartAndEndElements(context, syntaxTree, _logger);
+
+        // Make sure the selection starts on an element tag
+        if (startElementNode is null)
+        {
+            return SpecializedTasks.EmptyImmutableArray<RazorVSInternalCodeAction>();
+        }
+
+        if (!TryGetNamespace(context.CodeDocument, out var @namespace))
+        {
+            return SpecializedTasks.EmptyImmutableArray<RazorVSInternalCodeAction>();
+        }
+
+        var actionParams = CreateInitialActionParams(context, startElementNode, @namespace);
+
+        if (IsMultiPointSelection(context.Request.Range))
```

Comment:
Why does a multi-point selection need special handling? Surely we need to find the right nodes to process either way?

Reply:
True! This is fixed in later PR's but I'll go ahead and make this correction.